CMakeLists.txt

```diff
@@ -96,6 +96,17 @@ if (WITH_MPFR)
     set(HAVE_SYMENGINE_MPFR yes)
 endif()
 
+# Piranha
+set(WITH_PIRANHA no
+    CACHE BOOL "Build with Piranha")
+
+if (WITH_PIRANHA)
+    find_package(Piranha REQUIRED)
+    include_directories(${PIRANHA_INCLUDE_DIRS})
+    set(LIBS ${LIBS} pthread)
```

Comment:
@isuruf, perhaps instead of adding the dependencies like this, we should rather just
`` `
set(WITH_BOOST yes)
set(WITH_PTHREAD yes)
```
and then later have the usual `if (WITH_BOOST) ...`. The reason is that then the user can add `BOOST` but not `PIRANHA`. The same for `ARB` and its dependencies `MPFR` and `FLINT`. Now when we have `eval_mpfr`, you want to be able to sometimes link with `MPFR` but not with `ARB`.